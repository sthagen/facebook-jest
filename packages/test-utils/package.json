{
  "name": "@jest/test-utils",
  "version": "25.3.0",
  "private": true,
  "license": "MIT",
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "typesVersions": {
    "<3.8": {
      "build/*": [
        "build/ts3.4/*"
      ]
    }
  },
  "dependencies": {
    "@types/jest": "*",
    "@types/node": "*",
    "@types/semver": "^6.2.1",
    "ansi-regex": "^5.0.0",
    "ansi-styles": "^4.1.0",
    "semver": "^6.3.0"
  }
}
