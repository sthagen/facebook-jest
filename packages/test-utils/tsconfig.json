{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "build",
    // we don't want `@types/jest` to be referenced
    "types": ["node"]
  },
  "include": ["./src/**/*"],
  "references": [{"path": "../jest-types"}, {"path": "../pretty-format"}]
}
