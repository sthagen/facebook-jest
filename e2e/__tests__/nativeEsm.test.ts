/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {resolve} from 'path';
import wrap from 'jest-snapshot-serializer-raw';
import {onNodeVersions} from '@jest/test-utils';
import runJest, {getConfig} from '../runJest';
import {extractSummary} from '../Utils';

const DIR = resolve(__dirname, '../native-esm');

test('test config is without transform', () => {
  const {configs} = getConfig(DIR);

  expect(configs).toHaveLength(1);
  expect(configs[0].transform).toEqual([]);
});

// The versions vm.Module was introduced
onNodeVersions('^12.16.0 || >=13.2.0', () => {
  test('runs test with native ESM', () => {
    const {exitCode, stderr, stdout} = runJest(DIR, [], {
      nodeOptions: '--experimental-vm-modules',
    });

    const {summary} = extractSummary(stderr);

    expect(wrap(summary)).toMatchSnapshot();
    expect(stdout).toBe('');
    expect(exitCode).toBe(0);
  });
});
