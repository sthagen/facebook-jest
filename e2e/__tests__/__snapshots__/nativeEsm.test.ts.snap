// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on node >=16.9.0 support re-exports from CJS of dual packages 1`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /native-esm-deep-cjs-reexport.test.js/i."
`;

exports[`runs test with native ESM 1`] = `
"Test Suites: 1 passed, 1 total
Tests:       34 passed, 34 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /native-esm.test.js/i."
`;

exports[`supports top-level await 1`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /native-esm.tla.test.js/i."
`;
